### PR TITLE
feat(analytics): Add Confluence Analytics and Jira SLA metrics tools

### PR DESCRIPTION
**Pull request by meliezer124**
## Description

  Add 5 new MCP tools for analytics and workflow metrics:

  **Jira SLA Tools:**
  - `jira_get_issue_dates`: Extract dates and status transition history from issues
  - `jira_get_issue_sla`: Calculate cycle time, lead time, time in status, due date compliance with working hours support

  **Confluence Analytics Tools (Cloud only):**
  - `confluence_get_page_views`: Get view counts and unique viewers for pages
  - `confluence_get_page_analytics`: Calculate engagement score, view velocity, staleness metrics
  - `confluence_get_space_analytics`: Space-wide analytics with popular/stale page identification

  These tools enable workflow analysis and content health monitoring for Atlassian products.

  Fixes: #

  ## Changes

  - Add `jira/metrics.py` and `jira/sla.py` mixins for Jira date extraction and SLA calculations
  - Add `confluence/analytics.py` mixin for page and space analytics
  - Add Pydantic models for all new response types
  - Add 112 unit tests for new functionality
  - Add configuration options via environment variables (working hours, default metrics)
  - Update README with new tools documentation
  - Update `.env.example` with configuration examples

  ## Testing

  - [x] Unit tests added/updated
  - [x] Integration tests passed
  - [x] Manual checks performed: Tested with real Atlassian Cloud instance in Cursor IDE

  ## Checklist

  - [x] Code follows project style guidelines (linting passes).
  - [x] Tests added/updated for changes.
  - [x] All tests pass locally.
  - [x] Documentation updated (if needed).

  🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/mcp_atlassian/jira/sla.py [hunk 1]
```diff
@@ -0,0 +1,649 @@
...
+        if not issue_dates.created:
+            return LeadTimeMetric(
+                value_minutes=0,
+                formatted="0m",
+                is_resolved=False,
+            )
+
+        end_time = issue_dates.resolution_date or datetime.now(
+            tz=issue_dates.created.tzinfo
+        )
```
Comment (sooperset) [inline src/mcp_atlassian/jira/sla.py:311]: Using `issue_dates.created.tzinfo` can cause mismatch with the SLA config timezone.

Example: If an issue was created in UTC but SLA config uses `America/New_York`, working hours calculations will be incorrect.

Suggested fix:
```python
end_time = issue_dates.resolution_date or datetime.now(tz=tz)
```

Where `tz` should be the timezone from SLAConfig.

File: src/mcp_atlassian/jira/sla.py [hunk 2]
```diff
@@ -0,0 +1,649 @@
...
+                            change.entered_at,
+                            change.exited_at,
+                            working_hours_only,
+                            sla_config,
+                        )
+                        status_minutes += minutes
+                        visit_count += 1
+                    elif change.status == summary.status and not change.exited_at:
+                        # Current status - calculate to now
+                        now = datetime.now(tz=change.entered_at.tzinfo)
```
Comment (sooperset) [inline src/mcp_atlassian/jira/sla.py:357]: Current status time calculation uses `change.entered_at.tzinfo`. Same issue as above - should use the configured SLA timezone.

Suggested fix:
```python
now = datetime.now(tz=tz)  # Use timezone from SLAConfig
```

File: src/mcp_atlassian/jira/config.py [hunk 3]
```diff
@@ -14,6 +14,61 @@
...
+        working_hours_only = os.getenv(
+            "JIRA_SLA_WORKING_HOURS_ONLY", "false"
+        ).lower() in ("true", "1", "yes")
+
+        working_hours_start = os.getenv("JIRA_SLA_WORKING_HOURS_START", "09:00")
+        working_hours_end = os.getenv("JIRA_SLA_WORKING_HOURS_END", "17:00")
+
+        # Working days (1=Monday, 7=Sunday)
+        working_days_str = os.getenv("JIRA_SLA_WORKING_DAYS", "1,2,3,4,5")
+        working_days = [int(d.strip()) for d in working_days_str.split(",")]
```
Comment (sooperset) [inline src/mcp_atlassian/jira/config.py:57]: No validation that `working_days` values are in range 1-7 (isoweekday).

Invalid input like `JIRA_SLA_WORKING_DAYS=0,8,9` will silently fail to match any day.

Suggested fix:
```python
working_days = [int(d.strip()) for d in working_days_str.split(",")]
invalid_days = [d for d in working_days if d < 1 or d > 7]
if invalid_days:
    raise ValueError(f"Invalid working days: {invalid_days}. Must be 1-7 (Mon-Sun)")
```

File: src/mcp_atlassian/jira/sla.py [hunk 4]
```diff
@@ -0,0 +1,649 @@
...
+        if not issue_dates.resolution_date:
+            return ResolutionTimeMetric(
+                calculated=False,
+                reason="Issue not resolved",
+            )
+
+        # Find first "In Progress" transition
+        first_in_progress = None
+        for change in issue_dates.status_changes:
+            if change.status.lower() in ("in progress", "in development", "working"):
```
Comment (sooperset) [inline src/mcp_atlassian/jira/sla.py:458]: Only recognizes "In Progress", "In Development", "Working". Jira supports custom workflows, so statuses like "Code Review", "Blocked", or localized names will be ignored.

Suggested improvements:
1. Use status category API - Jira provides "To Do", "In Progress", "Done" categories
2. Or make the "in progress" status list configurable

File: src/mcp_atlassian/confluence/analytics.py [hunk 5]
```diff
@@ -0,0 +1,867 @@
...
+        previous_end = (now - timedelta(days=period_days)).strftime("%Y-%m-%d")
+
+        # Get current period views
+        try:
+            adapter = self._analytics_adapter
+            current_data = adapter.get_content_views(page_id, from_date=current_start)
+            current_views = current_data.get("count", 0)
+        except Exception as e:
+            logger.debug(f"Error getting current period views for {page_id}: {e}")
+            current_views = 0
```
Comment (sooperset) [inline src/mcp_atlassian/confluence/analytics.py:299]: All exceptions are silently caught and 0 is returned. Auth errors, network errors, rate limiting are all hidden, causing:
- Very difficult debugging
- Misleading metrics (API failure shown as "0 views")

Suggested fix:
```python
except HTTPError as http_err:
    if http_err.response.status_code in [401, 403]:
        raise  # Propagate auth errors
    logger.warning(f"Failed to get views for {page_id}: {http_err}")
    current_views = 0
except Exception as e:
    logger.warning(f"Unexpected error for {page_id}: {e}")
    current_views = 0
```

File: src/mcp_atlassian/confluence/analytics.py [hunk 6]
```diff
@@ -0,0 +1,867 @@
...
+        for page in all_pages:
+            page_id = page.get("content", {}).get("id") or page.get("id")
+            page_title = page.get("content", {}).get("title") or page.get("title")
+
+            if not page_id:
+                continue
+
+            try:
+                # Get view data
+                views_response = self.get_page_views(
```
Comment (sooperset) [inline src/mcp_atlassian/confluence/analytics.py:708]: Makes individual API calls for each page in the space. 500 pages = 500+ HTTP requests.

This causes:
- Very slow execution (can take minutes)
- Vulnerable to rate limiting
- Puts load on Confluence server

Suggested improvements:
1. Use the already implemented `batch_get_page_views()`
2. Or add result caching
3. Or use parallel requests (asyncio)

Example:
```python
page_ids = [str(p.get("content", {}).get("id") or p.get("id")) for p in all_pages]
batch_results = self.batch_get_page_views(page_ids, from_date=from_date)
```

File: src/mcp_atlassian/confluence/analytics.py [hunk 7]
```diff
@@ -0,0 +1,867 @@
...
+            errors=errors,
+            from_date=from_date,
+            to_date=to_date,
+        )
+
+    # =========================================================================
+    # Phase 4: Page Analytics (Calculated Metrics)
+    # =========================================================================
+
+    def _calculate_engagement_score(
```
Comment (sooperset) [inline src/mcp_atlassian/confluence/analytics.py:220]: The derived metrics (`engagement_score`, `staleness`, `view_velocity`) are calculated using arbitrary formulas:
- `view_score * 0.4 + viewer_score * 0.3 + recency_score * 0.3`
- Expected baseline: `period_days * 2`, `period_days * 0.5`
- Recency decay: 5 points per day

**Why remove:**
1. MCP is an API that provides data to LLMs
2. Just provide raw data (views, viewers, last_viewed) and let the LLM interpret it based on context
3. An arbitrary "engagement score of 75" has unclear meaning and can confuse rather than help
4. These are not official Atlassian metrics

**Suggestion:**
- Keep only `get_page_views` (raw data: views, viewers, last_viewed)
- Remove derived metrics from `get_page_analytics` and `get_space_analytics`
- Or merge `get_page_analytics` into `get_page_views`

File: src/mcp_atlassian/jira/metrics.py [hunk 8]
```diff
@@ -0,0 +1,408 @@
...
+            start: Start datetime
+            end: End datetime
+
+        Returns:
+            Duration in minutes (rounded)
+        """
+        delta = end - start
+        return int(delta.total_seconds() / 60)
+
+    def _format_duration(self, minutes: int) -> str:
```
Comment (sooperset) [inline src/mcp_atlassian/jira/metrics.py:372]: `_format_duration()` method is identically implemented in both `metrics.py` and `sla.py`.

File: tests/unit/jira/test_sla.py [hunk 9]
```diff
@@ -0,0 +1,564 @@
+"""Tests for the Jira SLA mixin."""
```
Comment (sooperset) [inline tests/unit/jira/test_sla.py:1]: 97 tests cover basic scenarios well, but important edge cases are missing:

**Missing tests:**
1. **Timezone tests** - SLA calculations with non-UTC timezones (America/New_York, Asia/Seoul, etc.)
2. **DST transitions** - Working hours calculation during daylight saving time changes
3. **Config validation** - Invalid working_days (0, 8), invalid timezone strings
4. **Duplicate tests** - Duration formatting tests duplicated between `test_metrics.py` and `test_sla.py`

Please add these tests when fixing the timezone bugs.

**Comment by meliezer124** [comment]
Closing this PR as requested. The changes have been split into smaller PRs for easier review:

  1. PR #838  - `jira_get_issue_dates` tool
  2. PR #839  - `jira_get_issue_sla` tool
  3. PR #840  - `confluence_get_page_views` tool

  Thank you for the thorough review and feedback!
